SCT failing to bring up the cluster because rsyslog docker can't find SSH key

I tried to run on SCT master the following test:
```
export SCT_SCYLLA_VERSION=5.0.rc6
export SCT_IP_SSH_CONNECTIONS=public
export SCT_INTRA_NODE_COMM_PUBLIC=true
export SCT_CLUSTER_HEALTH_CHECK=false
export SCT_REGION_NAME=eu-north-1
export SCT_INSTANCE_PROVISION=spot
export SCT_RUN_DB_NODE_BENCHMARKS=false
export JOB_NAME=local_run
hydra run-test performance_regression_alternator_test.PerformanceRegressionAlternatorTest.test_write --backend aws --config nyh/alternator-throughput-1.yaml
```

(my nyh/alternator-throughput-1.yaml is the copy of the existing alternator throughput yaml with a zero removed from all the numbers).

After 25 minutes, the test ended with an unclear error about logs 

```
< t:2022-05-31 15:02:35,412 f:wait.py         l:68   c:sdcm.wait            p:ERROR > Wait for: wait for db logs: timeout - 300 seconds - expired
```

see the full log below. It appears that SCT didn't even finish configuring the nodes, and didn't get to run the monitoring - at the end I am left with a live monitoring node running nothing (I successfully sshed to it and checked).  But I don't understand what went wrong that caused sct to stop setting up the cluster.

[sct.log](https://github.com/scylladb/scylla-cluster-tests/files/8805945/sct.log)

I ssh'ed into one of the db nodes, and Scylla seems to be running fine there.

I removed all the garbage docker processes with
```
docker kill `docker ps -q`
docker rm `docker ps -q -a`
```
And the bug remains - I still get these dockers being restarted constantly.
I ran "docker logs" on one of them, and got:

```
[WARN ] SSH_TUNNEL_HOST will be deprecated in v2.0
[WARN ] SSH_TUNNEL_LOCAL will be deprecated in v2.0
[WARN ] SSH_TUNNEL_REMOTE will be deprecated in v2.0
[WARN ] SSH_HOSTUSER will be deprecated in v2.0
[WARN ] SSH_HOSTNAME will be deprecated in v2.0
[WARN ] SSH_HOSTPORT will be deprecated in v2.0
[ERROR] No SSH Key file found
```

So I think now I know what the bug is. I just don't know how to tell it where the SSH Key file is...

> I removed all the garbage docker processes with
> 
> ```
> docker kill `docker ps -q`
> docker rm `docker ps -q -a`
> ```
> 
> And the bug remains - I still get these dockers being restarted constantly. I ran "docker logs" on one of them, and got:
> 
> ```
> [WARN ] SSH_TUNNEL_HOST will be deprecated in v2.0
> [WARN ] SSH_TUNNEL_LOCAL will be deprecated in v2.0
> [WARN ] SSH_TUNNEL_REMOTE will be deprecated in v2.0
> [WARN ] SSH_HOSTUSER will be deprecated in v2.0
> [WARN ] SSH_HOSTNAME will be deprecated in v2.0
> [WARN ] SSH_HOSTPORT will be deprecated in v2.0
> [ERROR] No SSH Key file found
> ```
> 
> So I think now I know what the bug is. I just don't know how to tell it where the SSH Key file is...

working log of the autossh docker image is:
```
[WARN ] SSH_TUNNEL_HOST will be deprecated in v2.0
[WARN ] SSH_TUNNEL_LOCAL will be deprecated in v2.0
[WARN ] SSH_TUNNEL_REMOTE will be deprecated in v2.0
[WARN ] SSH_HOSTUSER will be deprecated in v2.0
[WARN ] SSH_HOSTNAME will be deprecated in v2.0
[WARN ] SSH_HOSTPORT will be deprecated in v2.0
Agent pid 9
Identity added: (stdin) (fruch@scylladb.com)
[INFO] Using autossh 1.4g
[INFO] Tunneling scyllaadm@13.53.198.169:5000 to 127.0.0.1:49156
> autossh -M 0 -N -o StrictHostKeyChecking=no -o ServerAliveInterval=10 -o ServerAliveCountMax=3 -o ExitOnForwardFailure=yes -t -t -R 5000:127.0.0.1:49156 -p 22 scyllaadm@13.53.198.169
Warning: Permanently added '13.53.198.169' (ECDSA) to the list of known hosts.
```

--- a/sdcm/utils/syslogng.py
+++ b/sdcm/utils/syslogng.py
@@ -53,8 +53,8 @@ class SyslogNGContainerMixin:  # pylint: disable=too-few-public-methods
         os.makedirs(logdir, exist_ok=True)
         LOGGER.debug("syslog-ng will store logs at %s", logdir)
         volumes = {
-            self.syslogng_confpath: {"bind": "/config/syslog-ng.conf", "mode": "ro"},
-            logdir: {"bind": "/var/log"},
+            self.syslogng_confpath: {"bind": "/config/syslog-ng.conf", "mode": "ro,z"},
+            logdir: {"bind": "/var/log", "mode": "z"},
         }
         username = getpass.getuser()
         return {

